Account/FxADeviceRegistrator.swift

```diff
@@ -0,0 +1,125 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Deferred
+import Shared
+
+private let log = Logger.syncLogger
+
+// The current version of the device registration, we use this to re-register
+// devices after we update what we send on device registration.
+private let DeviceRegistrationVersion = 1
+
+public enum FxADeviceRegistrationResult {
+    case RegisteredOrUpdated
+    case AlreadyRegistered
+}
+
+public class FxADeviceRegistrator {
+    public enum DeviceRegistratorError: MaybeErrorType {
```

Comment:
Oh nevermind it might be used since it gets passed along with the deferreds. Either way - probably makes more sense to pull this out of the class and put it above.